### ISSUE TITLE
Sourcemaps don't work when using base in src()

### ISSUE THREAD

**Issue by poelstra** [ivogabe/gulp-typescript#32]
We recently switched from gulp-tsc to gulp-typescript (mainly because gulp-tsc seems to be no longer maintained, and you have already switched to TS 1.1, nice!)

However, we ran into a problem with source maps. I've made a small test case that exhibits the problem.

I'm using the following directory structure:
```
gulpfile.js
lib/main.ts
lib/sub/sub.ts
```

gulpfile.js:
```js
var gulp = require("gulp");
var ts = require('gulp-typescript');
var sourcemaps = require('gulp-sourcemaps');

gulp.task('default', function() {
	return gulp
		.src('lib/**/*.ts', { base: "." }) // Note: it works correctly without the base
		.pipe(sourcemaps.init())
		.pipe(ts({ "module": "commonjs"	}))
		.js
		.pipe(sourcemaps.write())
		.pipe(gulp.dest('release/js'));
});
```

Now, when I run `gulp`, I get the following output:
```
[16:20:51] Using gulpfile c:\Source\testje\gulpfile.js
[16:20:51] Starting 'default'...
gulp-sourcemap-write: source file not found:c:\Source\testje\lib\sub\sub\sub.ts
[16:20:51] Finished 'default' after 577 ms
```

Apparently, there's an extra `sub\` in that path that shouldn't be there.
If I remove the `{ base: "." }` from `gulp.src()`, it works OK, but we need that base in our (more complex) scenario.

I've tried to debug it, and have implemented the following work-around in `lib/project.ts`, in the `emit`-lambda at line 310, I've replaced
```ts
			if (map) sourcemapApply(file, map);
```
by
```ts
			if (map) {
				// map.sources contains a list of source files, but
				// their paths are relative to what TypeScript thought
				// was a good common base, whereas the path should
				// apparently be relative to originalName's directory.
				// So 'rebase' the map.sources to that.
				var srcRoot = this.program.getCommonSourceDirectory();
				var originalNameDir = path.dirname(originalName);
				var mapObj = JSON.parse(map);
				mapObj.sources = mapObj.sources.map((filename: string): string => {
					return path.relative(originalNameDir, path.resolve(srcRoot, filename));
				});
				map = JSON.stringify(mapObj);
				sourcemapApply(file, map);
			}
```

This works-for-me(TM), but I'm not sure whether it's the right solution.
Any ideas?

**Comment by ivogabe** [comment]
Does it help setting the sourceRoot to `""` (an empty string)?

**Comment by Merott** [comment]
Setting `sourceRoot` to `""` didn't work for me, perhaps because of my weird directory structure. I had to write a function to calculate relative source root paths for my `ts` files, and use that as `sourceRoot`. 

However, I don't have any issues when using `gulp-sourcemaps` for my `scss` files (`gulp-sass`), which appear right next to the `ts` files and therefore have to deal with the exact same folder structure. The difference though is that the content of the `scss` source files is included in the generated `css` outputs. With `includeContent` of `gulp-sourcemaps` set to true (its default value), setting the `sourceRoot` is unnecessary. Why is it that `gulp-typescript` output doesn't include source contents? I don't see `includeContent` option being set to false anywhere, so what's going on?

**Comment by ivogabe** [comment]
Fixed in cebc725b8e34957a077c198162551eb62064dede. Note that if you use `base` and the common directory of all source files isn't equal to the base path, you have to use TypeScript 1.5.0 beta, since it uses the newly added option `rootDir`. If the base path is the common directory of all sources, it should work with 1.4 too.

The `sourceRoot` option has been disabled, and is now automatically set to the base path. If you need a different sourceRoot, you can specify that using the options object of gulp-sourcemaps.

The source map paths are now relative and sources are always included.

See also #52 for more info about the changes to source maps support.
